File: Level 1/달리기 경주/solution.py
def solution(players, callings):
    rank_map = {player: index for index, player in enumerate(players)}

    for calling_player in callings:
        current_rank = rank_map[calling_player]
        front_player = players[current_rank - 1]

        players[current_rank - 1], players[current_rank] = (
            players[current_rank],
            players[current_rank - 1],
        )

        rank_map[calling_player] -= 1
        rank_map[front_player] += 1

    return players

File: Level 1/달리기 경주/solution.js
function solution(players, callings) {
  const rankMap = new Map();

  players.forEach((player, index) => rankMap.set(player, index));

  callings.forEach((callingPlayer) => {
    const currentRank = rankMap.get(callingPlayer);
    const frontPlayer = players[currentRank - 1];

    [players[currentRank - 1], players[currentRank]] = [
      players[currentRank],
      players[currentRank - 1],
    ];

    rankMap.set(callingPlayer, currentRank - 1);
    rankMap.set(frontPlayer, currentRank);
  });

  return players;
}
